runtime/integration-tests/src/generic/cases/lp/mod.rs

```diff
@@ -0,0 +1,912 @@
+			Token::Uint(U256::from(DEFAULT_BALANCE * DECIMALS_6)),
+		]),
+	)
+	.unwrap();
+
+	// ------------------ Substrate Side ----------------------- //
+	// Create router
+	let (base_fee, _) =
+		env.parachain_state(<T as pallet_evm::Config>::FeeCalculator::min_gas_price);
+
+	let evm_domain = EVMDomain {
+		target_contract_address: sp_core::H160::from(env.deployed("router").address().0),
+		target_contract_hash: BlakeTwo256::hash_of(&env.deployed("router").deployed_bytecode),
+		fee_values: FeeValues {
+			value: sp_core::U256::zero(),
+			gas_limit: sp_core::U256::from(500_000),
+			gas_price: sp_core::U256::from(base_fee),
+		},
+	};
+
+	let axelar_evm_router = AxelarEVMRouter::<T>::new(
+		EVMRouter::new(evm_domain),
+		BoundedVec::<u8, ConstU32<MAX_AXELAR_EVM_CHAIN_SIZE>>::try_from(
+			EVM_DOMAIN.as_bytes().to_vec(),
+		)
+		.unwrap(),
+		sp_core::H160::from(env.deployed("router").address().0),
+	);
+
+	env.parachain_state_mut(|| {
+		assert_ok!(
+			pallet_liquidity_pools_gateway::Pallet::<T>::set_domain_router(
+				RawOrigin::Root.into(),
+				Domain::EVM(EVM_DOMAIN_CHAIN_ID),
+				DomainRouter::<T>::AxelarEVM(axelar_evm_router),
+			)
+		);
+	});
+
+	// AddCurrency
+	// * register in OrmlAssetRegistry
+	// * trigger `AddCurrency`
+	let usdc_address = env.deployed("usdc").address();
+	env.parachain_state_mut(|| {
+		register_asset::<T>("USD Coin", "USDC", 6, 10_000, usdc_address, USDC)
+	});
+	let dai_address = env.deployed("dai").address();
+	env.parachain_state_mut(|| {
+		register_asset::<T>("Dai Coin", "DAI", 18, 100_000_000_000_000, dai_address, DAI)
+	});
+	let frax_address = env.deployed("frax").address();
+	env.parachain_state_mut(|| {
+		register_asset::<T>(
+			"Frax Coin",
+			"FRAX",
+			18,
+			100_000_000_000_000,
+			frax_address,
+			FRAX,
+		)
+	});
+
+	env.parachain_state_mut(|| {
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_currency(
+			OriginFor::<T>::signed(Keyring::Alice.into()),
+			USDC
+		));
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_currency(
+			OriginFor::<T>::signed(Keyring::Alice.into()),
+			DAI
+		));
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_currency(
+			OriginFor::<T>::signed(Keyring::Alice.into()),
+			FRAX
+		));
+
+		process_outbound::<T>()
+	});
+
+	// Create 2x pools
+	// * single tranched pool A
+	// * double tranched pool B
+
+	// AddPool A
+	// AddTranche 1 A
+	// AllowInvestmentCurrency 1
+	// AllowInvestmentCurrency 2
+	// AllowInvestmentCurrency 3
+
+	// AddPool B
+	// AddTranche 1 B
+	// AddTranche 2 B
+	// AllowInvestmentCurrency 1
+	// AllowInvestmentCurrency 2
+	// AllowInvestmentCurrency 3
+
+	// ------------------ EVM Side ----------------------- //
+
+	// Deploy LP and more for both pools and all currencies
+
+	// TODO: Does panic... Have we ever tested that? Building a block would be super
+	// useful to flush the pending of the evm side
+	// env.__priv_build_block(1);
+	env
+}
+
+pub fn register_asset<T: Runtime>(
```

Comment:
Regarding the above comment, see `register_currency()` from `generic/utils/currency.rs`